<head></head>
<body>
 <script>


/*----------------------------------Error Handling----------------------------------*/

/*
try: wrap suspicious code that may throw an error in try block.
     The try statement allows us to define a block of code to be tested for errors while it is being executed.

catch: write code to do something in catch block when an error occurs. 
       The catch block can have parameters that will give you error information. 
       Catch block is used to log an error or display specific messages to the user.

finally: finally block will always be executed regardless of the occurrence of an error. 
         The finally block can be used to complete the remaining task or reset variables that might have changed before error occurred in try block.

throw: the throw statement allows us to create a custom error. 
       We can through a string, number, boolean or an object. 
       Use the throw statement to throw an exception. 
       When you throw an exception, expression specifies the value of the exception. Each of the following throws an exception:    
*/




// try 
// {
//   let lastName = 'Yetayeh'
//   let fullName = fistName + ' ' + lastName
// } 
// catch (err) 
// {
//   confirm(err);
// }
// finally
// {
//     console.log("This is Finally")
// }





// let x = prompt('Enter a number: ')

// try 
// {
// x = Number(x)
// if (x == '') throw 'empty response'
// if (isNaN(x)) throw 'not a number'
// if (x < 5) throw 'too low'
// if (x > 10) throw 'too high'
// } 
// catch (err) 
// {
// console.log(err)
// } 
// finally
// {
//     console.log("Number is: " + x)
// }





/*----------------------------------Error Type----------------------------------*/



// ReferenceError: An illegal reference has occurred. A ReferenceError is thrown if we use a variable that has not been declared.

// let firstName = 'Asabeneh'                               // lastname is not defined
// let fullName = firstName + ' ' + lastName
// console.log(fullName)



// SyntaxError: A syntax error has occurred

// let square = 2 x 2;                                      // "*" comes insted of "x"
// console.log(square)




// TypeError: A type error has occurred

// let num = 20                                             //string method cant be applied to numbers
// console.log(num.toUpperCase())







</script>
</body>